### PR TITLE
Allow injection in IFrames + dynamic IFrames

### PR DESCRIPTION
**Pull request by sponnet**
When working with DAPPS that open Iframes - web3 is not injected there. These small changes fix this, and inject a web3 instance in every iframe that gets opened on page-start , and also on dynamically generated iframes.

We needed this for out TrellEth project 

https://twitter.com/sponnet/status/845100360540794880
https://press.swarm.city/meet-trelleth-73affcb82d02#.zhzcq2nhq
https://github.com/trelleth 

I hope this could get modified in the main MetaMask product. It would open up much possibilities !

### CODE REVIEW

**Review by GriffGreen — APPROVED** [review]
Looks pretty straight forward :-)

File: app/scripts/contentscript.js [hunk 1]
```diff
@@ -73,6 +73,6 @@ function isAllowedSuffix (testCase) {
   if (doctype) {
     return doctype.name === 'html'
   } else {
-    return false
+    return true
```
Comment (flyswatter) [inline app/scripts/contentscript.js]: This line would attempt to inject in any file without a doctyoe. What types of files are those? Surely HTML is already covered with line 74?

File: app/manifest.json [hunk 2]
```diff
@@ -51,7 +51,7 @@
         "scripts/contentscript.js"
       ],
       "run_at": "document_start",
-      "all_frames": false
+      "all_frames": true
```
Comment (flyswatter) [inline app/manifest.json]: This looks simple enough.

**Review by danfinlay — CHANGES_REQUESTED** [review]
Curious about a change to the default injection logic. We just modified this so it would stop injecting Inc xml, and this looks like it might hit other files.

File: app/scripts/contentscript.js (hunk 1 above)
Comment (sponnet) [inline app/scripts/contentscript.js]: Iframes apparently return null as a doctype in this function.

Comment (sponnet) [inline app/scripts/contentscript.js]: I'll try to make and add an example site URL  that opens the different situations in where iframes could be opened. Then you can check that they effectively don't send along a doctype. But I agree that this injection logic is there for a reason. If XML files actually send a doctype , they shouldn't be affected by this change tho.

Comment (sponnet) [inline app/scripts/contentscript.js]: I'll try to make and add an example site URL  that opens the different situations in where iframes could be opened. Then you can check that they effectively don't send along a doctype. But I agree that this injection logic is there for a reason. If XML files actually send a doctype , they shouldn't be affected by this change tho.

Comment (kingflurkel) [inline app/scripts/contentscript.js]: Sponnet likes to say things twice :)

Comment (sponnet) [inline app/scripts/contentscript.js]: The importance of TrellEth cannot be overstated :)

**Comment by kumavis** [comment]
@sponnet @kingflurkel can you tell us more about your use case, and TrellEth?

**Comment by kingflurkel** [comment]
https://press.swarm.city/meet-trelleth-73affcb82d02
@kumavis

**Comment by kingflurkel** [comment]
@kumavis We create this as a trello powerup, a sort of plugin. It works by injecting customizable iframes in trello when needed.

File: app/scripts/contentscript.js (hunk 1 above)
Comment (kumavis) [inline app/scripts/contentscript.js]: `document.doctype` has really poor support in general it seems

**Comment by sponnet** [comment]
The use of iframes in the trello power ups is mandatory, it's part of how their plugin architecture works.

Besides that IMHO it would make sense to look for a better implementation of the isAllowedSuffix function that also detects the mime type in an iframe. ( Ex opening an XML file in an iframe ) 

This pr works when the main window provides the doctype , then it only injects in HTML doctype. I haven't noticed mupliple injections in non iframe containing pages using this change.

**Comment by flyswatter** [comment]
Fixes #1223

**Comment by kumavis** [comment]
I'm willing to merge this and wait for push back from the community

**Comment by kumavis** [comment]
maybe we can add an iframe detection? `top !== window` or something

http://stackoverflow.com/questions/326069/how-to-identify-if-a-webpage-is-being-loaded-inside-an-iframe-or-directly-into-t

**Review by danfinlay — APPROVED** [review]
Alright, I can't think of any more situations this will go poorly in, let's try it, and see if anyone complains.

If you guys make some ethereum-enabled Trello powerup, we might have to switch to Trello.

**Comment by flyswatter** [comment]
It seems like our continuous integration tests aren't able to test off of someone else's remote?  I'm going to pull this & resubmit as a separate PR.

**Comment by flyswatter** [comment]
Merged in #1271.  Weird CI issues prevented merging this PR.

**Pull request merged**